### PR TITLE
rbrowser: Add support for DAG-ordered profiles with browser flags

### PR DESCRIPTION
**Pull request by kalbasit**
Add support for declarative browser profiles with support for browser flag per profiles. The profiles are shown in an order built using home-manager's excellent DAG type and library.  

I've also added the `--autodetect` flag to scan each browser's profile directory to append any detected profile to the list shown to rofi.

This PR requires https://github.com/nix-community/home-manager/pull/1673. I have not added that to the flake as I expect our soxincfg to pin to it instead.

### CODE REVIEW

File: modules/programs/rbrowser/default.nix [hunk 1]
```diff
@@ -1,20 +1,60 @@
...
+          name_profile = builtins.head (builtins.tail (builtins.split "^(.*)@(.*)$" dagName));
+        in
+        assert lib.asserts.assertMsg hasAt ''The dagName must be of the format browser@profile, got "${dagName}"!'';
+        assert lib.asserts.assertMsg (builtins.length name_profile == 2) "The dagName must be of the format browser@profile.";
+        builtins.elemAt name_profile index;
+    in
+    { dagName, ... }:
+    {
+      options = {
+        enable = mkOption {
```
Comment (rissson) [inline modules/programs/rbrowser/default.nix:21]: Do we still need this option? The user could simply remove the entry to disable it.

File: pkgs/rbrowser/default.nix [hunk 2]
```diff
@@ -1,202 +1,262 @@
...
 , runtimeShell
-}:
+, stdenvNoCC
+, writeScriptBin
 
-assert withBrave || withChromium || withFirefox;
+, browsers ? [
+    { name = "chromium"; profile = "personal"; flags = [ ]; }
+    { name = "brave"; profile = "personal"; flags = [ ]; }
+    { name = "firefox"; profile = "personal"; flags = [ ]; }
```
Comment (rissson) [inline pkgs/rbrowser/default.nix:14]:
```suggestion
    { name = "chromium"; profile = "default"; flags = [ ]; }
    { name = "brave"; profile = "default"; flags = [ ]; }
    { name = "firefox"; profile = "default"; flags = [ ]; }
```

File: modules/programs/rbrowser/default.nix [hunk 3]
```diff
@@ -1,20 +1,60 @@
...
-        type = types.bool;
-        description = ''
-          Whether to enable the ${name} integration with rbrowser.
-        '';
+  browserSubmodule =
+    let
+      getFromDagName = dagName: index:
+        let
+          hasAt = lib.stringAsChars (x: if x == "@" then x else "") dagName == "@";
+          name_profile = builtins.head (builtins.tail (builtins.split "^(.*)@(.*)$" dagName));
```
Comment (rissson) [inline modules/programs/rbrowser/default.nix:12]:
```suggestion
          nameProfile = builtins.head (builtins.tail (builtins.split "^(.*)@(.*)$" dagName));
```
Maybe let's keep the same naming conventions everywhere?

File: pkgs/rbrowser/default.nix [hunk 4]
```diff
@@ -1,202 +1,262 @@
...
-          if [[ -d "''${i}" ]]; then
-            items="''${items}''${sep}chromium@$(basename "''${i}")"
-            sep="\n"
-          fi
+  rbrowser = writeScriptBin "rbrowser"
+    (
+      let
+        items = builtins.concatStringsSep "\n" (map (browser: ''"${browser.name}@${browser.profile}"'') browsers);
+
+        declare_flags = builtins.concatStringsSep "\n" (map
```
Comment (rissson) [inline pkgs/rbrowser/default.nix:39]:
```suggestion
        declareFlags = builtins.concatStringsSep "\n" (map
```
Maybe let's keep the same naming conventions everywhere?

**Review by rissson — CHANGES_REQUESTED** [review]

**Comment by rissson** [comment]
One additional comment, maybe we can make the `@` separator configurable?

File: modules/programs/rbrowser/default.nix (hunk 1 above)
Comment (kalbasit) [reply modules/programs/rbrowser/default.nix:21]: Probably not. I'll get rid of it.

**Review by rissson — APPROVED** [review]

**Pull request merged**